nipst/nipst_builder.go

```diff
@@ -0,0 +1,221 @@
+package nipst
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"sync"
+	"time"
+)
+
+type proof struct{}
+
+type postCommitment struct {
+	Root  common.Hash
+	Proof proof
+}
+
+func (p *postCommitment) Valid() bool {
+	return true
+}
+
+//todo: figure out which fields are needed
+type MembershipProof struct {
+	Root  common.Hash
+	Proof proof
+}
+
+func (p *MembershipProof) Valid() bool {
+	return true
+}
+
+//todo: poet proof
+type PoetProof struct {
+	Root  common.Hash
+	Proof proof
+}
+
+func (p *PoetProof) Valid() bool {
+	return true
+}
+
+type PoetService interface {
+	SendPostProof(root common.Hash, p proof) error
```

Comment:
Perhaps `SubmitCommitment`, and a plain `[]byte` as an argument, so that the hashing logic will be in this layer (and not delegated to `PoetService`.

Reply:
I'd refrain from using primitive types that do not describe the parameter received